src/emqx_auth_http_app.erl

```diff
@@ -116,10 +115,13 @@ translate_env() ->
                         [] -> Acc;
                         Env ->
                             URL = proplists:get_value(url, Env),
-                            #{host := Host0,
-                              port := Port,
-                              path := Path} = uri_string:parse(URL),
-                            Host = get_addr(Host0),
+                            #{host := Host,
+                              path := Path} = URIMap = uri_string:parse(URL),
+                            {ok, PoolOpts} = application:get_env(?APP, pool_opts),
+                            Port = maps:get(port, URIMap, case proplists:get_value(ssl, PoolOpts, []) of
```

Comment:
#{host := Host,
                              path := Path,
                              scheme := Scheme} = URIMap = uri_string:parse(add_default_scheme(URL)),
                            Port = maps:get(port, URIMap, case Scheme of
                                      "https" -> 443;
                                      _ -> 80
                                  end),


add_default_scheme("http://" ++ URL) ->
    URL;
add_default_scheme("https://" ++ URL) ->
    URL;
add_default_scheme(URL) ->
    "http://" ++ URL.